x-pack/plugins/transform/public/app/sections/transform_management/components/transform_list/transform_list.tsx

```diff
@@ -317,14 +335,20 @@ export const TransformList: FC<TransformListProps> = ({
     </EuiFlexGroup>
   );
 
+  const handleSearchOnChange: EuiSearchBarProps['onChange'] = (search) => {
+    setQuery(search);
+    updatePageState({ queryText: search.queryText });
+  };
+
   const search = {
     toolsLeft: transformSelection.length > 0 ? renderToolsLeft() : undefined,
     toolsRight,
-    onChange: setQuery,
+    onChange: handleSearchOnChange,
     box: {
       incremental: true,
     },
     filters: transformFilters,
+    query: query?.queryText,
```

Comment:
Not sure if it's related to any changes here, but the filter buttons for status and health aren't working for me. Whatever I select, I get no matches. Only 'mode' works as expected.

<img width="357" alt="Screenshot 2024-08-02 at 16 13 29" src="https://github.com/user-attachments/assets/7d825c71-0b0a-4d47-b7ae-bbced646a7ac">

Reply:
Fixed in dd8e5b7726f0d3d110f6432bcab8825dfc12f05b